lib/sequelize.js

```diff
@@ -1145,7 +1145,7 @@ module.exports = (function() {
             return transaction.commit().then(function () {
               resolve(result);
             });
-          }).catch(function (err) {
+          }).then(null, function (err) {
```

Comment:
True, this style preserves original funcitonality where error in commit code path will be caught by the same error handler.